services/reproducible-central/reproducible-central.service.js

```diff
@@ -0,0 +1,91 @@
+import Joi from 'joi'
+import { BaseJsonService, pathParams } from '../index.js'
+
+const schema = Joi.object({
+  color: Joi.string().required(),
+  message: Joi.string().required(),
+}).required()
+
+export default class ReproducibleCentral extends BaseJsonService {
+  static category = 'dependencies'
+
+  static route = {
+    base: 'reproducible-central/a',
+    pattern: ':groupId/:artifactId/:version',
+  }
+
+  static openApi = {
+    '/reproducible-central/a/{groupId}/{artifactId}/{version}': {
+      get: {
+        summary: 'Reproducible Central',
+        parameters: pathParams(
+          {
+            name: 'groupId',
+            example: 'org.apache.maven',
+          },
+          {
+            name: 'artifactId',
+            example: 'maven-core',
+          },
+          {
+            name: 'version',
+            example: '3.9.9',
+          },
+        ),
+      },
+    },
+  }
+
+  static defaultBadgeData = {
+    label: 'Reproducible Builds',
+    labelColor: '1e5b96',
+  }
+
+  static render({ message, color }) {
+    return {
+      message,
+      color,
+    }
+  }
+
+  async fetch({ groupId, artifactId, version }) {
+    return this._requestJson({
+      schema,
+      url: `https://jvm-repo-rebuild.github.io/reproducible-central/badge/artifact/${groupId.replace(/\./g, '/')}/${artifactId}/${version}.json`,
+    })
+  }
+
+  async fetchIndex({ groupId, artifactId, version }) {
+    const { res } = await this._requestFetcher({
+      url: `https://jvm-repo-rebuild.github.io/reproducible-central/badge/artifact/${groupId.replace(/\./g, '/')}/${artifactId}/index.html`,
+    })
+    if (res.statusCode === 404) {
+      return { message: 'unknown ga', color: 'orange' } // unknown project ga, whatever version
+    } else {
+      return { message: version, color: 'grey' } // unknown version for a known project ga
```

Comment:
What are we trying to communicate here?
If I'm expecting a metric back and I'm getting back a version number that seems like an unexpected behaviour to me.
I'm aware that you have a specific web page you want to build, but if we're going to add this, I would want to make sure that we're providing something that makes sense to a generalised userbase. This seems to me like it should just be a simple error message. Some shorter words to say "This package hasn't been analysed by this service" or whatever.

Reply:
people ask for score for an artifactId, and de-facto it's a project that is rebuilt: that's a fact in Java/JVM projects, where one project builds many artifactIds at the same time
this leads to that unusual result in this badge: "only" unknown version is less unknown than unknown project

Reply:
Sounds like we really we want 2 errors: One for "project not found" and one error for "version not found" then.

Reply:
implementing the difference as errors is not an issue if you tell me how this looks as code, please :)
does it force red, or can the 2 levels of color be kept? (not a showstopper)

Reply:
For the error responses, we should actually throw them as exceptions (which will render an error badge image, but we don't explicitly set a colour).

So the "project not found" and "version not found" errors should be NotFound exceptions e.g:

```js
throw new NotFound({ prettyMessage: 'project not found' })
```

and your "snapshots are not analysed" error should be an `InvalidParameter` exception.

Reply:
now that I see the Shields approach, I think I'll change my data feed to change the approach
I'll just need some time, because travelling for 2 weeks